code/datums/components/omen.dm

```diff
@@ -0,0 +1,73 @@
+/**
+  * omen.dm: For when you want someone to have a really bad day
+  *
+  * When you attach an omen component to someone, they start running the risk of all sorts of bad environmental injuries, like nearby vending machines randomly falling on you,
+  * or hitting your head really hard when you slip and fall, or... well, for now those two are all I have. More will come.
+  *
+  * Omens are removed once the victim is either maimed by one of the possible injuries, or if they receive a blessing (read: bashing with a bible) from the chaplain.
+  */
+/datum/component/omen
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	/// Whatever's causing the omen, if there is one. Destroying the vessel won't stop the omen, but we destroy the vessel (if one exists) upon the omen ending
+	var/obj/vessel
+
+/datum/component/omen/Initialize(silent=FALSE, vessel)
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	var/mob/person = parent
+	if(!silent)
+		to_chat(person, "<span class='warning'>You get a bad feeling...</span>")
+	src.vessel = vessel
+
+/datum/component/omen/Destroy(force, silent)
+	if(vessel)
+		vessel.visible_message("<span class='warning'>[vessel] burns up in a sinister flash, taking an evil energy with it...</span>")
+	return ..()
+
+/datum/component/omen/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/check_accident)
+	RegisterSignal(parent, COMSIG_LIVING_STATUS_KNOCKDOWN, .proc/check_slip)
+	RegisterSignal(parent, COMSIG_ADD_MOOD_EVENT, .proc/check_bless)
+
+/datum/component/omen/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_LIVING_STATUS_KNOCKDOWN, COMSIG_MOVABLE_MOVED, COMSIG_ADD_MOOD_EVENT))
+
+/**
+  * check_accident() is called each step we take
+  *
+  * While we're walking around, roll to see if there's any environmental hazards (currently only vending machines) on one of the adjacent tiles we can trigger.
+  * We do the prob() at the beginning to A. add some tension for /when/ it will strike, and B. (more importantly) ameliorate the fact that we're checking up to 5 turfs's contents each time
+  */
+/datum/component/omen/proc/check_accident(atom/movable/our_guy)
+	if(!prob(15))
+		return
+	for(var/t in get_adjacent_open_turfs(our_guy))
+		var/turf/the_turf = t
+		for(var/obj/machinery/vending/darth_vendor in the_turf)
+			if(darth_vendor.tiltable)
+				darth_vendor.tilt(our_guy)
+				qdel(src)
+				return
+
+/// If we get knocked down, see if we have a really bad slip and bash our head hard
+/datum/component/omen/proc/check_slip(mob/living/our_guy, amount)
+	if(amount <= 0 || !prob(50))
```

Comment:
``!prob(50)`` :smirk:

Reply:
i swear when i first wrote this it wasn't a 50, it made sense then